### PR TITLE
ProfileScreen Changes

### PR DESCRIPTION
**Pull request by kumarwhocodes**
Implemented MVVM in ProfileScreen and created profileState sealed class to pass the event

### CODE REVIEW

**Comment by basakjeet08** [comment]
Issues :- 

- The ProfileScreen is having a instance of ViewModel in it which is not recommended. Instead pass the user data and the action functoin in the parameters 
- The profileAction function in View Model seems to have some redundant imports fix those !!

File: app/src/main/java/in/iot/lab/teacherreview/feature_teacherlist/presentation/state/ProfileState.kt
Comment (dead8309) [inline app/src/main/java/in/iot/lab/teacherreview/feature_teacherlist/presentation/state/ProfileState.kt:1]: This should be ProfileActions

File: app/src/main/java/in/iot/lab/teacherreview/feature_bottom_navigation/navigation/HomeNavGraph.kt [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
 import `in`.iot.lab.teacherreview.feature_teacherlist.presentation.screen.ProfileScreen
+import `in`.iot.lab.teacherreview.feature_teacherlist.presentation.stateholder.ProfileScreenViewModel
 
 /**
  * Navigation Graph : It contains all the Different Routes in the bottom Navigation
  */
 @Composable
 fun HomeNavGraph(navController: NavHostController) {
 
+    val viewModel :ProfileScreenViewModel = hiltViewModel()
```
Comment (dead8309) [inline app/src/main/java/in/iot/lab/teacherreview/feature_bottom_navigation/navigation/HomeNavGraph.kt:21]: This should either be placed in parameters of `HomeNavGraph` func. or in the content block of `ProfileRoute`

File: app/src/main/java/in/iot/lab/teacherreview/feature_bottom_navigation/navigation/HomeNavGraph.kt [hunk 2]
```diff
@@ -42,7 +47,13 @@ fun HomeNavGraph(navController: NavHostController) {
             // Profile Bottom Navigation Options
             composable(
                 BottomNavRoutes.ProfileRoute.route,
-                content = { ProfileScreen(navController = navController) }
+                content = { ProfileScreen(
+                    navController = navController,
+                    profileAction = viewModel::profileAction,
+                    userPhoto = viewModel.currentUser.collectAsState().value?.photoUrl.toString(),
+                    userEmail = viewModel.currentUser.collectAsState().value?.email.toString(),
+                    userUserName = viewModel.currentUser.collectAsState().value?.username.toString())
```
Comment (dead8309) [inline app/src/main/java/in/iot/lab/teacherreview/feature_bottom_navigation/navigation/HomeNavGraph.kt:55]: Extract out the currentUser state in a variable and use it in the following fields

File: app/src/main/java/in/iot/lab/teacherreview/feature_bottom_navigation/navigation/HomeNavGraph.kt [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
 import `in`.iot.lab.teacherreview.feature_teacherlist.presentation.screen.ProfileScreen
+import `in`.iot.lab.teacherreview.feature_teacherlist.presentation.stateholder.ProfileScreenViewModel
 
 /**
  * Navigation Graph : It contains all the Different Routes in the bottom Navigation
  */
 @Composable
-fun HomeNavGraph(navController: NavHostController) {
-
+fun HomeNavGraph(navController: NavHostController, profileVm: ProfileScreenViewModel) {
```
Comment (dead8309) [inline app/src/main/java/in/iot/lab/teacherreview/feature_bottom_navigation/navigation/HomeNavGraph.kt:19]: this should be 
`profileVm: ProfileScreenViewModel = hiltViewModel()`

File: app/src/main/java/in/iot/lab/teacherreview/feature_bottom_navigation/screens/HomeScreen.kt [hunk 4]
```diff
@@ -30,7 +33,8 @@ internal fun HomeScreen(navController: NavHostController = rememberNavController
                 .fillMaxSize()
                 .padding(bottom = it.calculateBottomPadding()),
         ) {
-            HomeNavGraph(navController = navController)
+            HomeNavGraph(navController = navController,
+                profileVm = profileVm)
```
Comment (dead8309) [inline app/src/main/java/in/iot/lab/teacherreview/feature_bottom_navigation/screens/HomeScreen.kt:37]: No need to create and pass vm instance, when the function parameter have a default value

**Pull request merged**